### PR TITLE
Blog_app processing data

### PR DESCRIPTION
**Pull request by firstpj**
In this pull request, I perform the following activities :

Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).

Set up associations between models.
[Use models to insert data](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#use-models-to-insert-data)

Familiarized myself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
Opened rails console for your project.

[Created custom methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#create-custom-methods)

My models include some additional methods.

Users
A method that returns the 3 most recent posts for a given user.

Posts
A method that updates the posts counter for a user.
A method that returns the 5 most recent comments for a given post.

Comments
A method that updates the comments counter for a post.

Likes
A method that updates the likes counter for a post.

Fix linter issues
Update readme

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, counter_cache: :comments_counter, optional: true
```
Comment (mckent05) [inline app/models/comment.rb:3]: - You have done a good job so far, but there is just one issue with your project, please remove the counter_cache from your project. The [counter_cache](https://scoutapm.com/blog/how-to-start-using-counter-caches-in-rails) functions similarly to the counter function you implemented, hence I get an extra value for the counter when I create a model. Please fix this 🙏

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, counter_cache: :likes_counter
```
Comment (mckent05) [inline app/models/like.rb:3]: - You have done a good job so far, but there is just one issue with your project, please remove the counter_cache from your project. The [counter_cache](https://scoutapm.com/blog/how-to-start-using-counter-caches-in-rails) functions similarly to the counter function you implemented, hence I get an extra value for the counter when I create a model. Please fix this 🙏

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: :posts_counter
```
Comment (mckent05) [inline app/models/post.rb:2]: - You have done a good job so far, but there is just one issue with your project, please remove the counter_cache from your project. The [counter_cache](https://scoutapm.com/blog/how-to-start-using-counter-caches-in-rails) functions similarly to the counter function you implemented, hence I get an extra value for the counter when I create a model. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @firstpj, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job setting up the associations between your models 🙏 🙏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @firstpj,

## APPROVED 🟢

## Highlights
✔️ Associations feature

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**